### ISSUE TITLE
Profile info on https://www.instagram.com/

### ISSUE THREAD

**Issue by LevPasha** [instabot-py/instabot.py#10]
Hi all! Does anyone know how to get the details of your profile on the site? For example who follow me and whom I follow, who like your photo. Without these data, the bot will not be complete.

Now i can follow anyone, but i can't unfollow someone because i can't get following list.

Any ideas?

**Comment by ThomasMW** [comment]
As far as I can tell there is no way to view the following list on the site, just the number of followers and following.

I'm guessing only their API provides this information.

**Comment by taktikal17** [comment]
@LevPasha I write on the file .txt, the user id that i followed. 
For my unfollow function, I read the the id on my .txt, and for the double check, I write again on another file .txt the user unfollowed ( you can get errors sometimes, was not able to unfollow..). I can compare between the two .txt and run again the script to unfollow the users that i couldn't unfollow the first time. 
Actually the only possibility to get the follower or following list is with webstagram, iconosquare, ink361.com.

Let's me know, if you can get the following list!

**Comment by LevPasha** [comment]
@taktikal17, hah! THX for http://ink361.com/. I think i can get following and follower lists soon.

**Comment by taktikal17** [comment]
@LevPasha, sounds great! 
If you can manage it, it means that we can get the follower list of every instagram users, and follow their followers! :100: 

I can't wait to see how you do it!! ( I need to learn it! :+1: )

**Comment by LevPasha** [comment]
@taktikal17 It is for you 38349f065a785f12fe3180b920e32ea2e27454ed 
Go to `userinfo.py`, uncomment 132-148 and look how it work.
I think you can integrate it in `instabot.py` yourself (fork your resolution).

**Comment by taktikal17** [comment]
@LevPasha Thanks!!
I am testing, it is just perfect!! :+1: 
I print all the id in a file:

`with open('following.txt', 'a') as f:
                            f.write("%s" %  u["id"].split("-")[1])
                            f.write("\n")
                            f.close()`

And I just need to read this file to follow the people in the instabot.py! 

Love it!! :100:

**Comment by LevPasha** [comment]
@taktikal17 i think you don't need file. Just call `UserInfo` class somewhere in `InstaBot` class. Get what you need, and use it.

Like:
```python
# in instabot.py
from userinfo.py import UserInfo
...
def login(self):
    self.user_data_list = UserInfo()
    self.user_data_list.search_user(user_name=self.user_login)
    self.user_data_list.get_followers()
    self.user_data_list_followers = sefl.user_data_list.followers

...
    # somewhere later:
    follow = self.follow(self.user_data_list_followers[0]["id"])
```
Don't check this code, this is just example.

**Comment by taktikal17** [comment]
Yes I tried it, but instagram blocks the number of following that you can get by day.
And when your run ui.get_follower, you always start with the follower[o].

You can't start again where you stopped the last time in the follower list..

The only thing boring with the file is you need to check which users you managed to follow with the bot, to delete them to the file in order to run again the file..

**Comment by taktikal17** [comment]
I wrote a script to compare your followers list with the followers list of another user and to write a new file with the users that you can follow. You can do the same between your follower and following list. :+1: 

' def difference(self):
        with open('yourfollower.txt', 'r+') as f1:
            print ("open")
            array1 = []
            for line in f1:
                array1.append(line)
        with open('user1follower.txt', 'r+') as f2:
            print ("open")
            array2 = []
            for line in f2:
                array2.append(line)
                new_list = []
 new_list = []
        for element in array2:
            if not element in array1:
                new_list.append(element)
                print (element)
                print (new_list)
                with open('needtofollow.txt', 'a') as f:
                    f.write("%s" % int(element))
                    f.write("\n")
                    f.close()'

**Comment by LevPasha** [comment]
@taktikal17 you can compare whom you want to like, with `ui.followers` list.

**Comment by taktikal17** [comment]
Currently, I am using the unfollow function, it works well. It takes a break sometimes, with an error 403. 
I already unfollowed 405 users in 5h.

Do you let your script running all the day? Or just couples of hours a day?

**Comment by LevPasha** [comment]
@taktikal17 i have own server, and script running 24/7, that make 1000 like at day. It work like this about month. Not use follow or comments for now.

**Comment by taktikal17** [comment]
Ok I was worry about using too much the bot, actually it is fine.
I think I am going to run the script for the likes and comments on my server, and I will keep the follow/unfollow on my command line.

Thank again for this script @LevPasha !

**Comment by taktikal17** [comment]
By the way @LevPasha, How to do you implement the `bot = InstaBot()` to run the like script 24/7 in order to make it starting again with the last posts ?

[https://github.com/LevPasha/instabot.py/issues/11](url), maybe start again at the part 2 of your screenshot.

Thanks!

**Comment by LevPasha** [comment]
@taktikal17 i use it like this:
```python
bot = InstaBot("tag_list" : my_tag_list) 
bot.auto_mod()
``` 
Loop: take media by random tag, then like some photos (1 to `max_like_for_one_tag`), when it end (`max_like_for_one_tag` or 20 media) start new loop.

For tag i like, it is no problem - people add hundreds of media in minute. If you have some not so popular tags - it is no solution for `auto_mod()`.

**Comment by taktikal17** [comment]
Ok, I get it. Thanks @LevPasha !

**Comment by kaxil** [comment]
Ya I managed to do that, but the bot seems to fail, while Unfollowing as @taktikal17 mentioned

**Comment by yohn89** [comment]
Suggestion:
Saving the cookie data to local file, so that the bot can login directly (with out a http POST request) when restarting the program.

**Comment by LevPasha** [comment]
@yohn89 hi, i think it is not very useful. You can write your code, and fork.

**Comment by 179skr** [comment]
Sorry for the oftopic, but how has everyone coded there follow someone elses followers bot. I don't even know how to code it. I'm trying to follow someone elses followers from a list and unfollow them after 24 hours. Thanks

**Comment by yohn89** [comment]
@179skr  Maybe you need to create a database or a file to store some infomation about that list,marking the timestamp when you follow some one,validating the timestamp when you unfollow some one.

**Comment by 179skr** [comment]
@yohn89 How do i code that? i'm a newbie so i don't know how to code this into the bot. lol thanks again.

**Comment by 179skr** [comment]
@taktikal17 Did you manage to build the follow bot?

**Comment by taktikal17** [comment]
@179skr You need to get the id of the user you want to follow ( UserInfo() , you print them in a .txt file.

You change your follow function to read the id in the .txt file and when you succeed to follow the user, you  print again the id in a new .txt file. And the same for the unfollow function. By the way, 24h it is not enough mainly during the working days. Keep them one week. 

@LevPasha I love your updates! Thanks!

**Comment by 179skr** [comment]
@taktikal17 could you email me the code your using please? I cannot seem to get this to work. My email is ameilakham12@gmail.com. Thank you.

**Comment by Moderndayfreak** [comment]
@LevPasha How many times are you running a cron job on your server? Thanks!

**Comment by GithubProgress** [comment]
Implemented in 1.2
